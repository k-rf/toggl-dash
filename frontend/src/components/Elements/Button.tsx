import { Button as MuiButton, ButtonProps as MuiButtonProps } from "@mui/material";

type Props = Pick<
  MuiButtonProps,
  "children" | "color" | "disabled" | "fullWidth" | "onClick" | "variant"
>;

export const Button = (props: Props) => {
  return <MuiButton {...props} />;
};
