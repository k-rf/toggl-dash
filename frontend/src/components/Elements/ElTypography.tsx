import { Typography, TypographyProps } from "@mui/material";

type Props = Pick<TypographyProps, "children" | "variant" | "color" | "sx">;

export const ElTypography = (props: Props) => {
  return <Typography {...props} />;
};
