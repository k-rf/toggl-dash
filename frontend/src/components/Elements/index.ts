export * from "./ElButton";
export * from "./ElPaper";
export * from "./ElTextField";
export * from "./ElAccordion";
