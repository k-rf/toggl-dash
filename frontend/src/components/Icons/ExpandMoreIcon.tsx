import { ExpandMore } from "@mui/icons-material";
import { SvgIconProps } from "@mui/material";

type Props = Pick<SvgIconProps, "color" | "fontSize">;

export const ExpandMoreIcon = (props: Props) => {
  return <ExpandMore {...props} />;
};
