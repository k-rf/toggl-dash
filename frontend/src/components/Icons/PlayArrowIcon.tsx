import { PlayArrow } from "@mui/icons-material";
import { SvgIconProps } from "@mui/material";

type Props = Pick<SvgIconProps, "color" | "fontSize">;

export const PlayArrowIcon = (props: Props) => {
  return <PlayArrow {...props} />;
};
