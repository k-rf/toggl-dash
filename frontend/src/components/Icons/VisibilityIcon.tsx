import { Visibility } from "@mui/icons-material";

export const VisibilityIcon = () => {
  return <Visibility />;
};
