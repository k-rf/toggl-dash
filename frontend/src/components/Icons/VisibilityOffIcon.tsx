import { VisibilityOff } from "@mui/icons-material";
import { SvgIconProps } from "@mui/material";

type Props = Pick<SvgIconProps, "color" | "fontSize">;

export const VisibilityOffIcon = (props: Props) => {
  return <VisibilityOff {...props} />;
};
