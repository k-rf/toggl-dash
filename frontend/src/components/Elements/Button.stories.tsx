import { Box, Stack } from "@mui/material";
import type { ComponentMeta, ComponentStoryObj } from "@storybook/react";

import { Button } from "./Button";

type Meta = ComponentMeta<typeof Button>;
type Story = ComponentStoryObj<typeof Button>;

export default { title: "Elements/Button", component: Button } as Meta;

export const Default: Story = {
  args: {
    children: "ボタン",
    color: "primary",
  },
};

export const Variant: Story = {
  ...Default,
  args: {
    ...Default.args,
  },
  render: (args) => {
    return <Buttons {...args} />;
  },
};

export const Disabled: Story = {
  ...Default,
  args: {
    ...Default.args,
    disabled: true,
  },
  render: (args) => {
    return <Buttons {...args} />;
  },
};

export const Colors: Story = {
  ...Default,
  args: {
    ...Default.args,
    fullWidth: true,
  },
  render: (args) => {
    return (
      <Stack direction={"column"} spacing={2}>
        <Buttons {...args} color="primary" width={120}>
          primary
        </Buttons>
        <Buttons {...args} color="secondary" width={120}>
          secondary
        </Buttons>
        <Buttons {...args} color="success" width={120}>
          success
        </Buttons>
        <Buttons {...args} color="warning" width={120}>
          warning
        </Buttons>
        <Buttons {...args} color="error" width={120}>
          error
        </Buttons>
        <Buttons {...args} color="info" width={120}>
          info
        </Buttons>
      </Stack>
    );
  },
};

const Buttons = (args: Story["args"] & { width?: number }) => {
  const { width, ...props } = args;
  return (
    <Stack direction={"row"} spacing={2}>
      <Box width={width}>
        <Button {...props} variant="contained" />
      </Box>
      <Box width={width}>
        <Button {...props} variant="outlined" />
      </Box>
      <Box width={width}>
        <Button {...props} variant="text" />
      </Box>
    </Stack>
  );
};
