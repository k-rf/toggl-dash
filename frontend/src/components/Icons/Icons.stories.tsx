import { Box, Stack } from "@mui/material";
import type { Meta, StoryObj } from "@storybook/react";

import { ElPaper } from "../Elements";
import { ElTypography } from "../Elements/ElTypography";

import { ExpandMoreIcon } from "./ExpandMoreIcon";
import { PlayArrowIcon } from "./PlayArrowIcon";
import { VisibilityIcon } from "./VisibilityIcon";
import { VisibilityOffIcon } from "./VisibilityOffIcon";

export default {
  title: "Components/Icons",
  render: () => {
    return (
      <Box display="flex" flexWrap="wrap">
        <SxPaper title="Visibility">
          <VisibilityIcon />
        </SxPaper>
        <SxPaper title="VisibilityOff">
          <VisibilityOffIcon />
        </SxPaper>
        <SxPaper title="PlayArrow">
          <PlayArrowIcon />
        </SxPaper>
        <SxPaper title="ExpandMore">
          <ExpandMoreIcon />
        </SxPaper>
        {/* NOTE: プロダクトで利用するアイコンが増えたらここに追記する */}
      </Box>
    );
  },
} as Meta;

export const Default: StoryObj = {};

const SxPaper = ({ title, children }: { title: string; children: React.ReactNode }) => {
  return (
    <ElPaper
      sx={{
        display: "flex",
        width: 96,
        height: 48,
        justifyContent: "center",
        alignItems: "center",
        m: (theme) => theme.spacing(0, 2, 2, 0),
      }}
    >
      <Stack>
        <Box display="flex" justifyContent="center">
          {children}
        </Box>
        <ElTypography variant="caption">{title}</ElTypography>
      </Stack>
    </ElPaper>
  );
};
