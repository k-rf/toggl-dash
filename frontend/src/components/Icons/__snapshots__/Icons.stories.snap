// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Icons/Icons > Default 1`] = `
<div
  className="MuiBox-root css-2imjyh"
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-wnuo1l-MuiPaper-root"
  >
    <div
      className="css-nen11g-MuiStack-root"
    >
      <div
        className="MuiBox-root css-1l4w6pd"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="VisibilityIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
          />
        </svg>
      </div>
      <span
        className="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
      >
        Visibility
      </span>
    </div>
  </div>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-wnuo1l-MuiPaper-root"
  >
    <div
      className="css-nen11g-MuiStack-root"
    >
      <div
        className="MuiBox-root css-1l4w6pd"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="VisibilityOffIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78 3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
          />
        </svg>
      </div>
      <span
        className="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
      >
        VisibilityOff
      </span>
    </div>
  </div>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-wnuo1l-MuiPaper-root"
  >
    <div
      className="css-nen11g-MuiStack-root"
    >
      <div
        className="MuiBox-root css-1l4w6pd"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="PlayArrowIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M8 5v14l11-7z"
          />
        </svg>
      </div>
      <span
        className="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
      >
        PlayArrow
      </span>
    </div>
  </div>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-wnuo1l-MuiPaper-root"
  >
    <div
      className="css-nen11g-MuiStack-root"
    >
      <div
        className="MuiBox-root css-1l4w6pd"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ExpandMoreIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </div>
      <span
        className="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
      >
        ExpandMore
      </span>
    </div>
  </div>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-wnuo1l-MuiPaper-root"
  >
    <div
      className="css-nen11g-MuiStack-root"
    >
      <div
        className="MuiBox-root css-1l4w6pd"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="UnfoldLessIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7.41 18.59 8.83 20 12 16.83 15.17 20l1.41-1.41L12 14l-4.59 4.59zm9.18-13.18L15.17 4 12 7.17 8.83 4 7.41 5.41 12 10l4.59-4.59z"
          />
        </svg>
      </div>
      <span
        className="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
      >
        UnfoldLess
      </span>
    </div>
  </div>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-wnuo1l-MuiPaper-root"
  >
    <div
      className="css-nen11g-MuiStack-root"
    >
      <div
        className="MuiBox-root css-1l4w6pd"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="UnfoldMoreIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 5.83 15.17 9l1.41-1.41L12 3 7.41 7.59 8.83 9 12 5.83zm0 12.34L8.83 15l-1.41 1.41L12 21l4.59-4.59L15.17 15 12 18.17z"
          />
        </svg>
      </div>
      <span
        className="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
      >
        UnfoldMore
      </span>
    </div>
  </div>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-wnuo1l-MuiPaper-root"
  >
    <div
      className="css-nen11g-MuiStack-root"
    >
      <div
        className="MuiBox-root css-1l4w6pd"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </div>
      <span
        className="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
      >
        Menu
      </span>
    </div>
  </div>
</div>
`;
